collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java

```diff
@@ -131,27 +131,27 @@ private boolean uploadOneSubmission(String urlString, String id, String instance
 
         File instanceFile = new File(instanceFilePath);
         ContentValues cv = new ContentValues();
-        Uri u = Uri.parse(urlString);
+        Uri submissionUri = Uri.parse(urlString);
```

Comment:
Making this final will let readers know the reference will not change.

Reply:
Ah, good ideas, @dcbriccetti! Am on it now.

Reply:
Never underestimate the cognitive load–reducing power of the `final`! 😁 
Especially over long distances.
I usually don’t use them over short distances.

Reply:
This one potentially does change if there is a redirect received from the HEAD request.

Reply:
Whoops, I missed the error message from A.S. when I tried it. Sorry.